cds-snc/cppd-medical-report-sails
# Switch session store to Postgres

# What this does

Swaps out our session store to Postgres instead of Redis. This is being done primarily to simplify our architecture/deployment.

Note that I've created a session.js model file - this is just so the session table required by the middleware gets generated on sails lift. This should be replaced by a simple migration file once we have our migrations workflow.

## To test

- Check out this branch
- Run `npm install` to get the new dependency
- Change (or create) your session adapter environment variable (in .env) to `SESSION_ADAPTER="connect-pg-simple"`
- Start the service
- Visit the start page

You should have a new `session` table in your database. In that table, you should see the session you just started.

Note that the session table that gets created automatically gets `createdAt` and `updatedAt` columns that we don't need - these get added by sails. As mentioned, this should just move to a migration, just using the model for now for simplicity.

The middleware author provides this .sql file for generating the table (for future reference):

```
CREATE TABLE "session" (
  "sid" varchar NOT NULL COLLATE "default",
	"sess" json NOT NULL,
	"expire" timestamp(6) NOT NULL
)
WITH (OIDS=FALSE);

ALTER TABLE "session" ADD CONSTRAINT "session_pkey" PRIMARY KEY ("sid") NOT DEFERRABLE INITIALLY IMMEDIATE;

CREATE INDEX "IDX_session_expire" ON "session" ("expire");
```
